faker/providers/person/pl_PL/__init__.py

```diff
@@ -725,30 +703,74 @@ def identity_card_number(self):
 
         return ''.join(str(character) for character in identity)
 
-    def pesel(self):
+    @staticmethod
+    def pesel_compute_check_digit(x):
+        checksum_values = [9, 7, 3, 1, 9, 7, 3, 1, 9, 7]
+        return sum(int(a) * b for a, b in zip(list(x), checksum_values)) % 10
+
+    def pesel(self, date_of_birth=None, sex=None):
         """
         Returns 11 characters of Universal Electronic System for Registration of the Population.
         Polish: Powszechny Elektroniczny System Ewidencji Ludności.
 
         PESEL has 11 digits which identifies just one person.
-        Month: if person was born in 1900-2000, december is 12. If person was born > 2000, we have to add 20 to month,
-        so december is 32.
-        Person id: last digit identifies person's sex. Even for females, odd for males.
+        pesel_date: if person was born in 1900-2000, december is 12. If person was born > 2000, we have to add 20 to
+        month, so december is 32.
+        pesel_sex: last digit identifies person's sex. Even for females, odd for males.
 
         https://en.wikipedia.org/wiki/PESEL
         """
+        if date_of_birth is None:
+            date_of_birth = self.generator.date_of_birth()
 
-        birth = self.generator.date_of_birth()
+        pesel_date = '{year}{month:02d}{day:02d}'.format(
+            year=date_of_birth.year, day=date_of_birth.day,
+            month=date_of_birth.month if date_of_birth.year < 2000 else date_of_birth.month + 20)
+        pesel_date = pesel_date[2:]
 
-        year_pesel = str(birth.year)[-2:]
-        month_pesel = birth.month if birth.year < 2000 else birth.month + 20
-        day_pesel = birth.day
-        person_id = self.random_int(1000, 9999)
+        pesel_core = ''.join(list(map(str, [self.random_digit() for _ in range(3)])))
```

Comment:
can we avoid the calls to `list` here and use generators all the way?

Reply:
Yes, we can.